webroot/src/network/userApi/interceptors.ts

```diff
@@ -14,9 +14,18 @@ import { authStorage, tokens } from '@/app.config';
  * @return {AxiosInterceptor} Function that amends the outgoing client API request.
  */
 export const requestSuccess = () => async (requestConfig) => {
-    const authToken = authStorage.getItem(tokens.staff.AUTH_TOKEN);
-    const authTokenType = authStorage.getItem(tokens.staff.AUTH_TOKEN_TYPE);
-    const { headers } = requestConfig;
+    const licenseeUserEndPoints = ['/v1/provider-users/me'];
+    const { headers, url } = requestConfig;
+    let authToken;
+    let authTokenType;
+
+    if (licenseeUserEndPoints.includes(url)) {
+        authToken = authStorage.getItem(tokens.licensee.ID_TOKEN);
+        authTokenType = authStorage.getItem(tokens.licensee.AUTH_TOKEN_TYPE);
+    } else {
+        authToken = authStorage.getItem(tokens.staff.AUTH_TOKEN);
+        authTokenType = authStorage.getItem(tokens.staff.AUTH_TOKEN_TYPE);
+    }
```

Comment:
Can you add a comment briefly explaining this? I _think_ the intent is to handle cases when someone is both a staff and a licensee?

Also, is there a conceptual reason it's called `ID_TOKEN` for the licensee and `AUTH_TOKEN` for the staff?

Reply:
I will.

Essentially, we have two different user pools that produce different unrelated auth tokens depending on if the user is a licensee or a staff user. Storing the tokens in different places and treating them as ignorant of each other is a knock-on effect of this, the "two app" approach and the edge cases that involve a user going through the sign in flow for both user types.

Certain endpoints are designed to be called by staff users and certain endpoints are designed to be called by licensee users. We need to send the correct token (determined by which endpoint we are hitting) up with the call.

I dont know the underlying reason why the licensee endpoints want the Id token and the staff endpoints want the auth token. Landon or Justin would be better able to answer that question.